packages/3d-web-client-core/src/character/LocalController.ts

```diff
@@ -67,71 +72,78 @@ export class LocalController {
     private readonly keyInputManager: KeyInputManager,
     private readonly cameraManager: CameraManager,
     private readonly timeManager: TimeManager,
-  ) {}
+  ) {
+    setInterval(() => this.update.bind(this), 3000);
+  }
 
   public update(): void {
     if (!this.model?.mesh || !this.model?.animationMixer) return;
     if (!this.thirdPersonCamera) this.thirdPersonCamera = this.cameraManager.camera;
 
-    const movementKeysPressed = this.keyInputManager.isMovementKeyPressed();
-    const forward = this.keyInputManager.isKeyPressed("w");
-    const backward = this.keyInputManager.isKeyPressed("s");
-    const left = this.keyInputManager.isKeyPressed("a");
-    const right = this.keyInputManager.isKeyPressed("d");
+    const { forward, backward, left, right, run, jump, anyDirection, conflictingDirection } =
+      this.keyInputManager;
 
-    this.inputDirections = { forward, backward, left, right };
-    this.jumpInput = this.keyInputManager.isJumping();
-    this.runInput = this.keyInputManager.isShiftPressed();
+    this.forward = forward;
+    this.backward = backward;
+    this.left = left;
+    this.right = right;
+    this.run = run;
+    this.jump = jump;
+    this.anyDirection = anyDirection;
+    this.conflictingDirections = conflictingDirection;
 
-    if (movementKeysPressed) {
+    this.targetSpeed = this.run ? this.maxRunSpeed : this.maxWalkSpeed;
+    this.speed += ease(this.targetSpeed, this.speed, 0.07);
+
+    this.rayCaster.set(this.model.mesh.position, this.vectorDown);
+    const hit = this.rayCaster.intersectObjects([this.collisionsManager.colliders]);
+    if (hit.length > 0) this.currentHeight = hit[0].distance;
+
+    if (anyDirection) {
       const targetAnimation = this.getTargetAnimation();
       this.model.updateAnimation(targetAnimation, this.timeManager.deltaTime);
     } else {
       this.model.updateAnimation(AnimationState.idle, this.timeManager.deltaTime);
     }
 
-    if (Object.values(this.inputDirections).some((v) => v)) {
-      this.updateRotation();
-    }
+    if (this.anyDirection) this.updateRotation();
 
     for (let i = 0; i < this.collisionDetectionSteps; i++) {
       this.updatePosition(this.timeManager.deltaTime / this.collisionDetectionSteps, i);
     }
 
-    if (this.model.mesh.position.y < 0) {
-      this.resetPosition();
-    }
+    if (this.model.mesh.position.y < 0) this.resetPosition();
     this.updateNetworkState();
   }
 
   private getTargetAnimation(): AnimationState {
```

Comment:
There seems to be a bug where the air and jumping animations don't get triggered if you're stationary when you jump

Reply:
The problem was in [this line](https://github.com/mml-io/3d-web-experience/pull/3/commits/f2ea998e0d88868d960f90992fc7ea86d58e9d94#diff-1af65b2e2e07611732279ba5207d01b6de1781d6e48358b549879cdb88a0efcdR100) preventing the `getTargetAnimation` method to be called if there were no directional keys pressed.

This is now fixed on my local branch and ready to be pushed in my next PR